daft/io/parquet.py

```diff
@@ -43,16 +45,34 @@ def read_parquet(
     returns:
         DataFrame: parsed DataFrame
     """
+
     if isinstance(path, list) and len(path) == 0:
         raise ValueError(f"Cannot read DataFrame from from empty list of Parquet filepaths")
 
-    plan = _get_tabular_files_scan(
-        path,
-        schema_hints,
-        ParquetSourceInfo(
-            io_config=io_config,
-            use_native_downloader=use_native_downloader,
-        ),
-        fs,
-    )
+    context = get_context()
+
+    if context.use_rust_planner:
+        plan = cast(
```

Comment:
Instead of this type casting, we could make the `DataFrame` constructor take a `Union[LogicalPlan, RustLogicalPlanBuilder]`.